rest/rest-server/src/test/java/org/apache/brooklyn/rest/entitlement/AuthenticateAnyoneSecurityProvider.java

```diff
@@ -0,0 +1,23 @@
+package org.apache.brooklyn.rest.entitlement;
+
+import javax.servlet.http.HttpSession;
+
+import org.apache.brooklyn.rest.security.provider.SecurityProvider;
+
+public class AuthenticateAnyoneSecurityProvider implements SecurityProvider {
```

Comment:
With `AnyoneSecurityProvider`, the user wasn't set. I needed to have authenticate impl as `user != null` before the user was actually set in the entitlements context.

Not sure if we could change `AnyoneSecurityProvider`, as that would presumably require username+password to always be passed in for all the other tests.